signer/core/api.go

```diff
@@ -430,6 +443,34 @@ func (api *SignerAPI) EcRecover(ctx context.Context, data, sig hexutil.Bytes) (c
 	return crypto.PubkeyToAddress(*rpk), nil
 }
 
+// SharedSecret returns the shared ECDH secret, salted and hashed.
+// TODO: Documentation in the Wiki
+func (api *SignerAPI) SharedSecret(ctx context.Context, otherKey, salt hexutil.Bytes) (hexutil.Bytes, error) {
+	// Get the other public key
+	otherPubKey, err := crypto.UnmarshalPubkey(otherKey)
+	if err != nil {
+		return nil, err
+	}
+	addr := crypto.PubkeyToAddress(otherPubKey)
+	// We make the request prior to looking up if we actually have the account, to prevent
+	// account-enumeration via the API
+	req := &SharedSecretRequest{Address: addr}
+	res, err := api.UI.ApproveSharedSecret(req)
+
+	if err != nil {
+		return nil, err
+	}
+	if !res.Approved {
+		return nil, ErrRequestDenied
+	}
+	// Look up the wallet containing the requested signer
+	account := accounts.Account{Address: addr.Address()}
+	wallet, err := api.am.Find(account)
+	if err != nil {
+		return nil, err
+	}
+}
```

Comment:
So the remainder of this function, should it invoke the `account.SharedSecret( some_addr)` ?